tests/functional/test_home.py

```diff
@@ -0,0 +1,62 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from selenium.common.exceptions import TimeoutException
+
+from ..pages.home import HomePage
+
+
+@pytest.mark.nondestructive
+def test_promos_are_present(base_url, selenium):
+    page = HomePage(base_url, selenium).open()
+    assert page.promos.length == 16
+
+
+@pytest.mark.nondestructive
+def test_tweet_is_present(base_url, selenium):
+    page = HomePage(base_url, selenium).open()
+    assert page.promos.tweet_is_present
+
+
+@pytest.mark.nondestructive
+def test_download_button_is_displayed(base_url, selenium):
+    page = HomePage(base_url, selenium).open()
+    assert page.download_button_is_displayed
+
+
+@pytest.mark.nondestructive
+def test_upcoming_events_are_displayed(base_url, selenium):
+    page = HomePage(base_url, selenium).open()
+    events = page.upcoming_events
+    assert events.is_next_event_displayed
+    assert events.is_events_list_displayed
+
+
+@pytest.mark.nondestructive
+def test_mozilla_newsletter_default_values(base_url, selenium):
+    page = HomePage(base_url, selenium).open()
+    page.mozilla_newsletter.expand_form()
+    assert '' == page.mozilla_newsletter.email
+    assert 'United States' == page.mozilla_newsletter.country
+    assert not page.mozilla_newsletter.privacy_policy_accepted
+
+
+def test_mozilla_newsletter_successful_sign_up(base_url, selenium):
+    page = HomePage(base_url, selenium).open()
+    success = page.mozilla_newsletter.success_url_slug
+    page.mozilla_newsletter.expand_form()
+    page.mozilla_newsletter.type_email('noreply@mozilla.com')
+    page.mozilla_newsletter.select_country('United Kingdom')
+    page.mozilla_newsletter.accept_privacy_policy()
+    page.mozilla_newsletter.click_sign_me_up()
+    assert success in selenium.current_url
```

Comment:
I think we should abstract the URL slug into the page object entirely. Perhaps a property such as `sign_up_successful` that returns a boolean based on this slug being in the URL.